src/main/java/Duke/SaveFile/DataSaver.java

```diff
@@ -0,0 +1,151 @@
+        String[] taskDetails = newTask.split("\\|");
+
+        if (!checkValidDetails(taskDetails)) {
+            throw new DukeException();
+        }
+
+        String type = taskDetails[0].trim();
+
+        switch(type) {
+        case "T":
+            String todoDescription = taskDetails[2].trim();
+            Todo addedTodo = new Todo(todoDescription);
+            addDoneStatus(addedTodo, taskDetails);
+            taskList.add(addedTodo);
+            break;
+        case "D":
+            String deadlineDescription = taskDetails[2].trim();
+            String deadlineBy = taskDetails[3].trim();
+            Deadline addedDeadline = new Deadline(deadlineDescription, deadlineBy);
+            addDoneStatus(addedDeadline, taskDetails);
+            taskList.add(addedDeadline);
+            break;
+        case "E":
+            String eventDescription = taskDetails[2].trim();
+            String eventAt = taskDetails[3].trim();
+            Event addedEvent = new Event(eventDescription, eventAt);
+            addDoneStatus(addedEvent, taskDetails);
+            taskList.add(addedEvent);
+            break;
+        default:
+            throw new DukeException();
+        }
+    }
+
+    public static boolean checkValidDetails(String[] taskDetails) {
+        String status = taskDetails[1].trim();
+        if (!status.equals(DONE_STATUS) && !status.equals(NOT_DONE_STATUS)) {
+            return false;
+        }
+
+        return taskDetails.length == 3 || taskDetails.length == 4;
+    }
+
+    public static void addDoneStatus(Task addedTask, String[] taskDetails) {
+        String status = taskDetails[1].trim();
+        if (status.equals(DONE_STATUS)) {
+            addedTask.markAsDone();
+        }
+    }
+
+    public static void manageSave(ArrayList<Task> taskList) {
+        try {
+            saveFileContents(taskList);
+        } catch (IOException ioException) {
+            DukeException.SaveIOException(ioException);
+        }
+    }
+
+    public static void saveFileContents(ArrayList<Task> taskList) throws IOException {
+        FileWriter writeFile = new FileWriter(FILE_PATH);
+        configureTask(taskList, writeFile);
+        writeFile.close();
+    }
+
+    public static void configureTask(ArrayList<Task> taskList, FileWriter writeFile) throws IOException {
+        for (Task task : taskList) {
+            StringBuilder parsedTask = new StringBuilder();
+            parseType(task, parsedTask);
+            parseStatus(task, parsedTask);
+            parseDescription(task, parsedTask);
+            writeFile.write(parsedTask + System.lineSeparator());
+        }
+    }
+
+    public static void parseType(Task task, StringBuilder parsedTask) {
+        String type = task.getType();
+        parsedTask.append(type).append(DIVIDER);
+    }
+
+    public static void parseStatus(Task task, StringBuilder parsedTask) {
+        if (task.getIsDone()) {
+            parsedTask.append("1").append(DIVIDER);
```

Comment:
You may want to define what "1" and "0" means, if it refers to done status, then do remember to use the constant you have defined.